### PR TITLE
Gianpiero/board print conn comp

### PR DESCRIPTION
**Pull request by GianpieroCea**
This PR aims to achieve the following:

- [ x] minimal command line representation of status of board
- [x ] random policy for player to take random choice
- [ ] Implementation of connected component data structure
- [ ] implementation of check if game is finished function

Currently there is some bug with the implementation of connected component but the skeleton should be there

### CODE REVIEW

**Comment by GianpieroCea** [comment]
I made the test conn comp test work by using  a hack.
Then added a new test to board._update_connected_components as there is something still wrong happening where not only the red graph is updated but also the blue one.

My idea is that if we can first make the test pass and get it working we can close the pr and make it nicer/more optimised later.

File: hexgame/src/unionfind.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        which is taken to be as the representative of the
+        disjoint set
+        """
+        parent = self._parents[a]
+        # father is a father of itself iff is root
+        # so while not reached root
+        while parent != self._parents[parent]:
+            # chase ancestors
+            parent = self._parents[parent]
+        # amortized code:
```
Comment (dinies) [inline hexgame/src/unionfind.py:57]: What are we doing here ?
I don't think this section is needed if the only thing that we want to do is to return the root of the tree to which a belongs.

**Review by dinies — COMMENTED** [review]
I will take some time to test this and maybe add some unit tests helpful to pinpoint the problem.

Meanwhile I have a question on `Unionfind::find()`

**Comment by GianpieroCea** [comment]
Now implemented the correct logic to check if game has finished, using the connected components. Ready to merge.

**Review by dinies — APPROVED** [review]
I don’t have time to review but I trust your judgement

**Pull request merged**